sphinx/application.py

```diff
@@ -318,7 +319,7 @@ def _log(self, message, wfile, nonl=False):
             wfile.flush()
         self.messagelog.append(message)
 
-    def warn(self, message, location=None, prefix='WARNING: '):
+    def warn(self, message, location=None, prefix='WARNING: ', type=None, subtype=None):
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Redefining built-in 'type'](https://www.codacy.com/app/georg/sphinx/file/2137828463/issues/source?bid=3149936&fileBranchId=3149940#l322)